File: Problem 1 Codes/1_claude_sol2.py

import sys


def find_range_bounds(target, value_set):
    lo = target
    hi = target
    while (lo - 1) in value_set:
        lo -= 1
    while (hi + 1) in value_set:
        hi += 1
    return lo, hi


def solve_case(n, x, queue):
    value_set = set(queue)
    lo, hi = find_range_bounds(x, value_set)
    front_val = queue[0]
    return "YES" if lo <= front_val <= hi else "NO"


def main():
    input_data = sys.stdin.read().split()
    pos = 0
    t = int(input_data[pos])
    pos += 1

    output_lines = []
    for _ in range(t):
        n = int(input_data[pos])
        x = int(input_data[pos + 1])
        pos += 2

        queue = list(map(int, input_data[pos: pos + n]))
        pos += n

        output_lines.append(solve_case(n, x, queue))

    print("\n".join(output_lines))


if __name__ == "__main__":
    main()

File: Problem 1 Codes/1_claude_sol2.java


import java.io.*;
import java.util.*;

class Main {

    private static long[] findRangeBounds(long target, HashSet<Long> valueSet) {
        long lo = target;
        long hi = target;
        while (valueSet.contains(lo - 1))
            lo--;
        while (valueSet.contains(hi + 1))
            hi++;
        return new long[] { lo, hi };
    }

    private static String solveCase(long frontVal, long x, HashSet<Long> valueSet) {
        long[] bounds = findRangeBounds(x, valueSet);
        long lo = bounds[0], hi = bounds[1];
        return (frontVal >= lo && frontVal <= hi) ? "YES" : "NO";
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StreamTokenizer st = new StreamTokenizer(br);
        st.nextToken();
        int t = (int) st.nval;

        StringBuilder out = new StringBuilder();

        for (int tc = 0; tc < t; tc++) {
            st.nextToken();
            int n = (int) st.nval;
            st.nextToken();
            long x = (long) st.nval;

            long[] queue = new long[n];
            HashSet<Long> valueSet = new HashSet<>(n * 2);
            for (int i = 0; i < n; i++) {
                st.nextToken();
                queue[i] = (long) st.nval;
                valueSet.add(queue[i]);
            }

            out.append(solveCase(queue[0], x, valueSet)).append('\n');
        }

        System.out.print(out);
    }
}